File: Data Structures & Algorithms/duplicate-integer/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        let numSet = new Set();

        for (let num of nums) {
            if (numSet.has(num)) {
                return true;
            }
            numSet.add(num);
        }

        return false;
    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        set<int> history = {};

        for (int i = 0; i < nums.size(); i++) {
            int num = nums[i];
            if (history.count(num)) {
                return num;
            }
            history.insert(num);
        }
    }
};
